### DISCUSSION TITLE
How can I recover cart after a user logout and login again?

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4387] [Q&A] [answered]
For example
- User A login
- User A add a product into the cart
- User A logout

I have stored the `cart.id` as a cookie. When User A logout I will remove that id as well. But the problem is, how can I recover that cart for User A? There seems not to be an endpoint that let me do that

**Comment by tsvetann** [comment, marked as answer]
https://github.com/medusajs/medusa/discussions/4311
